pkg/helm/actions/install_chart.go

```diff
@@ -5,38 +5,50 @@ import (
 	"helm.sh/helm/v3/pkg/chart"
 	"helm.sh/helm/v3/pkg/chart/loader"
 	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/client-go/util/retry"
 )
 
 func InstallChart(ns, name, url string, vals map[string]interface{}, conf *action.Configuration) (*release.Release, error) {
-	cmd := action.NewInstall(conf)
+	var release *release.Release
 
-	releaseName, chartName, err := cmd.NameAndChart([]string{name, url})
-	if err != nil {
-		return nil, err
-	}
-	cmd.ReleaseName = releaseName
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		cmd := action.NewInstall(conf)
 
-	cp, err := cmd.ChartPathOptions.LocateChart(chartName, settings)
-	if err != nil {
-		return nil, err
-	}
+		releaseName, chartName, err := cmd.NameAndChart([]string{name, url})
+		if err != nil {
+			return err
+		}
+		cmd.ReleaseName = releaseName
 
-	ch, err := loader.Load(cp)
-	if err != nil {
-		return nil, err
-	}
+		cp, err := cmd.ChartPathOptions.LocateChart(chartName, settings)
+		if err != nil {
+			return err
+		}
 
-	// Add chart URL as an annotation before installation
-	if ch.Metadata == nil {
-		ch.Metadata = new(chart.Metadata)
-	}
-	if ch.Metadata.Annotations == nil {
-		ch.Metadata.Annotations = make(map[string]string)
-	}
-	ch.Metadata.Annotations["chart_url"] = url
+		ch, err := loader.Load(cp)
+		if err != nil {
+			return err
+		}
+
+		// Add chart URL as an annotation before installation
+		if ch.Metadata == nil {
+			ch.Metadata = new(chart.Metadata)
+		}
+		if ch.Metadata.Annotations == nil {
+			ch.Metadata.Annotations = make(map[string]string)
+		}
+		ch.Metadata.Annotations["chart_url"] = url
+
+		cmd.Namespace = ns
+		release, err = cmd.Run(ch, vals)
+
+		// Cleanup failed release before re-Run
+		if err != nil {
+			UninstallRelease(name, conf)
```

Comment:
Can the return value of this line be discarded? I think we can because the error will be reflected on the next retry cycle anyway, and will cause next retry to fail. In addition, I don't want the error of this line obscure the error returned by `cmd.Run`.

Reply:
Might need opinions from other folks on this.